### ISSUE TITLE
Upload through app removes GPS data from images

### ISSUE THREAD

**Issue by AykutCevik** [nextcloud/android#6248]
### Steps to reproduce
1. Take a picture with a camera app which writes also GPS data into the picture
2. Ensure GPS data is written into the picture with different apps like file browser or gallery app
3. Upload this picture with the Nextcloud Android app version 3.11.1 via the share menu from gallery app
4. Download picture from Nextcloud instance with a browser and look at the GPS data of the file
5. See that the GPS data is missing or more likely set to zero

### Expected behaviour
- Pictures shouldn't lose their GPS data while uploaded through the app
![grafik](https://user-images.githubusercontent.com/1965251/84023644-be34d000-a988-11ea-90ad-45536866c31f.png)

### Actual behaviour
- Pictures are missing GPS data after upload
![grafik](https://user-images.githubusercontent.com/1965251/84023713-e3c1d980-a988-11ea-851a-5fee30530245.png)

### Can you reproduce this problem on https://try.nextcloud.com?
- Upload through the browser to Nextcloud works fine, just app removes the informations

### Environment data
Android version: 10 (OxygenOS 10.3.3)
Device model: OnePlus 6T
Stock or customized system: Stock
Nextcloud app version: 3.11.1
Nextcloud server version: 18.0.5
Reverse proxy: Nginx

#### Web server error log
```
Nothing worth to mention
```

#### Nextcloud log (data/nextcloud.log)
Only related to "Maps" app while trying to analyze the image since the GPS data is wrong.
```
Nothing worth to mention
```

I faced this issue since I saw that the pictures weren't showing up in the Maps app of Nextcloud. So this is not a new issue. Also I can surely say that the problem started before 20th of April, so also with a prior version. I can share pictures in a PM.

Appreciate any ideas for solving this issue.
Thank you for your great work!

**Comment by strugee** [comment]
>     3\. Upload this picture with the Nextcloud Android app version 3.11.1

How are you uploading the picture? Auto Upload? The "Upload files" button? The "Upload content from other apps" button?

> Stock or customized system: FALSE

Does this mean you have a stock system? (I.e. the version of Android that came with your phone)

> Reverse proxy: FALSE

I believe this question is asking what web server you access Nextcloud through.

**Comment by AykutCevik** [comment]
@strugee updated the issue.

Trying several other uploads, I found out that the issue may not be related to the Nextcloud app itself.

I've uploaded pictures through the share menu from both apps:
* OnePlus stock gallery app
* Simple gallery app (com.simplemobiletools.gallery.pro)

Both uploads resulted in setting the GPS data to zero.

Upload through a file browser like "Amaze" or through the Nextcloud app itself didn't change the GPS data.

Tricky one! I will contact the app maintainers and reference to this issue here.

**Comment by AykutCevik** [comment]
I have to reopen it again since I'm not quite sure how to debug this issue without brainstorming/information from here.

The "simple gallery" app provider said that they don't change the image beforehand. What could be the problem?

**Comment by AykutCevik** [comment]
Also to reference a comment which describes exactly the same thing: https://github.com/nextcloud/maps/issues/361#issuecomment-614571815

**Comment by strugee** [comment]
> Upload through a file browser like "Amaze" or through the Nextcloud app itself didn't change the GPS data.

How exactly did you upload in all of these apps? Did you use the "Share" button in Amaze? In Simple Gallery? In the stock gallery? In general the more information and specifics contained in a bug report, the better.

**Comment by AykutCevik** [comment]
@strugee as I said in point 3:
`via the share menu from gallery app`
This was the approach for all of the apps. The gallery app is stock, yes.

**Comment by tobiasKaminsky** [comment]
NC is not changing any metadata.
You should easily find the causing app via uploading with different apps.
If you can really reproduce it via NC please re-open.

**Comment by AykutCevik** [comment]
Sorry, but I have to reopen this. It is still an issue, if I share a picture through the gallery or files app, nextcloud removes/nulls the GPS data all the time: 
![grafik](https://user-images.githubusercontent.com/1965251/167653122-a6221b32-c69e-4a19-a556-0a406115a563.png)

**Comment by AykutCevik** [comment]
@tobiasKaminsky can we reopen this issue please? It prevents me from uploading images through gallery apps which is way more convenient than doing it through the Nextcloud app.

**Comment by fm-sys** [comment]
Stripping EXIF data is part of the android Q privacy feature set.

Starting with Android 10 (or 11?), apps need to explicitly specify that they want to receive Metadata.

So it's indeed a bug in the nextcloud client, and not the fault of speciffic gallery apps.

You can read more about it here:
- https://stackoverflow.com/questions/68442139/exif-location-data-removed-when-sharing-photo-via-intent
- https://developer.android.com/reference/android/provider/MediaStore#EXTRA_ACCEPT_ORIGINAL_MEDIA_FORMAT
- https://issuetracker.google.com/issues/128675076

**Comment by AykutCevik** [comment]
@AlvaroBrey any chance we get feedback on this issue since @fm-sys thankfully added more information?

**Comment by andyxheli** [comment]
Hopefully a resolution soon on this :)

**Comment by hubble14567** [comment]
I found a solution, not sure it works everywhere, but it fixed every upload done with the Nexcloud app (notably instant upload).
(I have a OnePlus7 with Android 11)

Go to your phone Settings -> Privacy -> Permission Manager -> Files and Media. Find Nexcloud in the list and make sure to "Allow management of all file". This works for me

![Screenshot_20220807-110434res](https://user-images.githubusercontent.com/70491689/183272038-ab64379b-0910-4143-952e-8218131048ed.jpg)

Previously this setting was on "Allow access to media only".

**Comment by andyxheli** [comment]
I get gps  data on my Samsung S21 + with auto uploads  on. It started to work. I have no idea what changed.

**Comment by hubble14567** [comment]
Tested some more. Funny thing is only instant upload and "Upload files" from the  "all app" menu is working. 
The share button in Gallery app (this app has the same auth as Nextcloud) does not work, nor the "Upload files" for the "shared" menu in the Nextcloud app.

**Comment by AykutCevik** [comment]
Uploading within the Nextcloud app already worked before in my case, it is cropping GPS data when you use the share functionality of other apps like gallery and files apps.
They are way better in showing, organizing and selecting files and images than the Nextcloud app which makes it hard to use the in-app upload of NC regularly.

**Comment by Poldi1977** [comment]
The problem also occurs under Android 13 on a Pixel 4.

**Comment by AykutCevik** [comment]
Clearly, this is a case of data loss! If I wouldn't have found out and would just have all my photos in the nextcloud instance I would never be able to get back my GPS data in the photos. How is this issue not getting more attention? How many users may be already affected by this?

**Comment by InfamousUser** [comment]
Nobody seems to care about these issues. And the amount of them is vast.

**Comment by strugee** [comment]
I would encourage anyone who cares about seeing this get solved to open a PR. There seems to be a path forward described in https://github.com/nextcloud/android/issues/6248#issuecomment-1147167524 which should help you.

**Comment by stephan-l** [comment]
Is there any hope to get this fixed? Without valid metadata auto upload is completly useless.

The solution in privacy dashboard is not usable as there are only two options on Pixel 7.

**Comment by tobiasKaminsky** [comment]
Can you test this: https://github.com/nextcloud/android/pull/11161#issuecomment-1345998605 ?

**Comment by AykutCevik** [comment]
Works for me @tobiasKaminsky

**Comment by stephan-l** [comment]
Seems to work for me too @tobiasKaminsky

**Comment by dbmaxpayne** [comment]
I just stubled upon this too.
Very frustrating as I had already deleted the originals on the phone when I was wondering where the GPS info is -.-
Either way, this works on Google Pixel 7 Pro https://github.com/nextcloud/android/issues/6248#issuecomment-1207315271

**Comment by AykutCevik** [comment]
@dbmaxpayne definitely! This data loss has to be addressed in a unit or integration test. I'm really wondering how at NC or nobody else faced this issue so far - maybe they don't know? People should check the maps feature in NC after upload to see if their photos are placed on the map or not.

**Comment by dbmaxpayne** [comment]
I mean I really didn't notice until now and look at how old this thread already is.
I only noticed, because my workflow has changed due to a new phone.
Back then I always copied the files off the phone over the InstantUpload ones to get rid of them on the phone and to circumvent any upload issues I sometimes had with my old Samsung S7. I always wondered why Nextcloud on Windows would upload the files again without noticing they are actually the same.
Now I know. They were never the same -.-

As I work quite a lot with the GPS data for example, I'd also vote for some sort of check in the Android app's code if it is allowed to upload the original file and display a warning if not.
However, I am not deep enough into Android programming as to suggest a fix myself.

Either way it's a weird behavior by Android to remove data without any notification at all.
Nextcloud seems to just be a victim here.

**Comment by whitemanthedj** [comment]
+1 on everything said. Would love to have gps data automatically uploaded with user permission. Media location not working from my auto upload setup from my Samsung S10. Have noticed for a while that my photos didn't show up on the map, but just getting around to pursuing a fix and the magical world of open source.

**Comment by rikrdo89** [comment]
have anyone found a fix to this issue for the Sealife client on android? 
It seems like no one is responding to the issues opened by users.  (https://github.com/haiwen/seadroid/issues/946)

**Comment by vitis586** [comment]
This is indeed a problem but it is not entirely problem with Nextcloud. I am also missing all location data on my auto uploaded files but when I tried sharing photos from other apps (ie. uploaded a photo as a file to Telegram and downloaded to my PC), I was missing the GPS data too. Not missing, but zeroed out as @AykutCevik shows. When I uploaded the same photo from Google Photos gallery, I have the GPS data present and accurate. The steps below helped me to keep the GPS data in photos.

**Is there a way for Nextcloud to ask for this specific permission?** As you can see the depth of the setting is quite substantial and it took me quite some time to find it.

This is a bit different on my Pixel 7 Pro - https://github.com/nextcloud/android/issues/6248#issuecomment-1207315271

Settings - Security and Privacy - Privacy - Permissions manager - Files - (small print on the bottom) See more apps that can access all files - Nextcloud - Allow access to manage all files

![obrazek](https://user-images.githubusercontent.com/55784861/232313968-b5c20d11-523c-4021-81cb-51f1b5e98ab3.png)

**Comment by brilthor** [comment]
I would suggest that the nextcloud client should be notifying users if it's not getting GPS data as part of autoupload and how to work around it until there is a transparent solution.  

I like many others found out about this data loss issue when looking at backups and trying to figure out why they were corrupted which is a terrible place to put a user in

**Comment by Cebrain** [comment]
Have to say this is a huge bug for me, that the location data is removed while upload.
It started for me in January 2023 that the data went missing, dont know exactly what changed.

I am really happy i still have the files on my phone so no location data was lost and i moved the files manually.

**Comment by nomandera** [comment]
Please consider family deployments, (non skilled users), when resolving.

Currently if the client permissions are not correct, images may be uploaded with missing metadata. I cannot trust my users to understand anything to do with permissions and I cannot rely on initial setup to persist as Android routinely suggest removes permissions from applications under its own control.

Currently in these scenarios my family photos result in silent data loss that I cannot mitigate. It would be better if image auto-uploads did not work at all unless the full necessary permissions were granted.  Please fail closed in all cases, even edge cases, where any data loss can occur.

In the interim please urgently add a client warning. I ended up here after hours of head scratching debugging why on earth some of our holidays photos are missing Lat/Long and it is certain many others will unknowingly be in the same boat.

**Comment by reox** [comment]
> It would be better if image auto-uploads did not work at all unless the full necessary permissions were granted.

I guess there are cases where you want auto upload but strip sensitive metadata (okay maybe not that often tbh, but you never know). 
I would vote for a solution that neither silently ignores the metadata nor never uploads anything without granting that permission. In my opinion, the best thing to do is to inform the user during initial setup and ask what to do.

> In the interim please urgently add a client warning. 

I sign that. I also spend half a day scratching my head what the heck was going on there...

**Comment by reox** [comment]
Is it possible that Android resets this permission after a while? I noticed that it suddenly stopped working and the permission to read all files was removed again.

**Comment by vitis586** [comment]
Mine permission are still the same since I set it up. Did you use Nextcloud app? Android removes permissions after some time of inactivity.

**Comment by reox** [comment]
I used it for the upload - does that not count as usage? but that might be the case.. I disabled the auto-removal for nextcloud now as well.

**Comment by nomandera** [comment]
Some info that I am not sure has been posted so far. I was hit with this bug but my initial symptoms were lack of auto-upload deleting the redundant copy on android on successful upload.

This eventually led me here to the real after I noticed some GEO tags missing from my families photos.

So purely anecdotally it appears that in my case at least the failure to upload a proper copy of the image also manifests as failure to delete.

Unfortunately I manually deleted some photos from my phone essentially losing the geographic information forever when I was sill under the mistaken impression this was a much simpler problem.

The client MUST alerts for missing permissions. The current situation of silent data loss is horrendous.

**Comment by vb0** [comment]
Can the client in the first place detect that it's getting sanitized/censored files from the OS? How could it tell that it didn't receive a "proper copy" if it didn't have access to the original? Sure, there might be something that can be done "out of band" in the API but it's highly unlikely as it's some logic to be explicitly programmed and it doesn't seem to be activity on this since 2020.  

Other than that I'd go one step further about alerting: the information should be present in **any file transfer notification**. Frankly the OS should be doing it in the first place when it starts to zero out stuff in your files, I don't understand why they don't do it as it seems to be a feature many people are very vocal about it, if Android caters to these even better, just pop up a notification "I just sanitized for you 1, 5, 200 files that had GPS location you didn't want to preserve/share/transfer/etc.". Let them feel better that Android is protecting them from themselves while letting also users that know better something is happening. But this is what it is, next best thing would be for NextCloud (and anything similar) to alert the user about this.

**Comment by AykutCevik** [comment]
> Can the client in the first place detect that it's getting sanitized/censored files from the OS?

It is not about detecting it, it is about requesting the sufficient rights from the apps perspective and warn a user if rights are missing to access the file properly with all the data.

**Comment by vb0** [comment]
> It is not about detecting it, it is about requesting the sufficient rights from the apps perspective and warn a user if rights are missing to access the file properly with all the data.  

That is just one way of detecting (or at least assuming) you aren't getting the complete file, but as I said it's some extra logic that for sure it isn't yet programmed (as NextCloud doesn't even know to request the right permissions yet AFAIK, never mind keeping track they weren't granted). My question was triggered by the other previous comment mentioning files that weren't uploaded properly not being deleted too, was this by chance or NextCloud somehow detected it isn't getting the originals?

**Comment by strugee** [comment]
> My question was triggered by the other previous comment mentioning files that weren't uploaded properly not being deleted too, was this by chance or NextCloud somehow detected it isn't getting the originals?

It was probably by chance. That behavior is likely triggered by an error being thrown, interrupting the upload process before it has a chance to delete the file on the local device. By contrast, this bug doesn't cause files to fail to upload - rather it causes the _wrong_ file (the sanitized/scrubbed version of the picture instead of the original) to be uploaded.

**Comment by nomandera** [comment]
Can i just clarify these responses because as I see it my logic HAS to be incorrect or this would be an absolutely horrendous critical bug that has been open for potentially 2+ years.

If it was `probably by chance` that the final file deletion was blocked by permissions on the device does this not mean that in some scenarios the NextCloud client is uploading files with incomplete meta and then deleting the copy with complete metadata i.e. losing data.

**Comment by DAveShillito** [comment]
I _think_ a recent update may have fixed the issue.
I just took a picture, and it auto uploaded and the version that arrived on my PC _did_ have the location in the metadata.

**Comment by troygeiger** [comment]
> I _think_ a recent update may have fixed the issue. I just took a picture, and it auto uploaded and the version that arrived on my PC _did_ have the location in the metadata.

I also just found that the location does auto-uploading on my Galaxy phone after discovering there is a location tag option in the camera app that was toggled off. Turned it on and sure enough, the location was in Nextcloud in the next picture I took.

**Comment by vb0** [comment]
Still happening on the latest non-beta from Play Store (3.26.0) on Pixel 7 with Android 13 (September update).

When testing it's worth sharing the following:  

* permissions, especially if you gave the special "All files access" one (it's a different UI for that). We know it works with that, and it's the current workaround.  
* which software version/branch and where it comes from (Play Store, F-Droid, side-loaded, etc.)  
* what's the phone manufacturer and which software is running (both as Android version/last update and if it's something like LineageOS).

**Comment by asm0dey** [comment]
Thanks @vitis586, your fix worked for me! Is there any way to resync all the data now to upload the information to nextcloud?

**Comment by Cebrain** [comment]
> Thanks @vitis586, your fix worked for me! Is there any way to resync all the data now to upload the information to nextcloud?

You could create a new folder on your phone with the problematic media files.
Enable Auto Upload for this folder with the Upload Setting "Upload all existing files" and "Overwrite existing files"

You could also do this on you normal camera folder, if you want reupload everything.

**Comment by vitis586** [comment]
> Thanks @vitis586, your fix worked for me! Is there any way to resync all the data now to upload the information to nextcloud?

You can do

**Comment by asm0dey** [comment]
@Cebrain I want to reupload all from my camera folder, but do not understand how, sorry :(

**Comment by Cebrain** [comment]
> @Cebrain I want to reupload all from my camera folder, but do not understand how, sorry :(

This is no problem, you only need to disable the auto upload for the Camera folder. Save the setting.
Then enable Auto Upload again for Camera Folder with this 2 settings "Upload all existing files" and "Overwrite existing files"

**Comment by mrcg** [comment]
Manual activation of "allow access to all files" in android settings for the nextcloud app solves the issue.
Anyhow about 98% off all my applications have "allow accesss to all files" activated by default.
This kind of mandatory setting should be set by default when installing the nextcloud app as this is also done from the majority of other android apps. I have been struggling a lot on this topic and lot ofther users might complain as well. Nextcloud is already great but it will become even better in usability by such change.

**Comment by joshtrichards** [comment]
> This kind of mandatory setting should be set by default when installing the nextcloud app 

@mrcg It already is. Well, it's requested at install time. The user can choose the other option on the pop-up though, which will obviously result in us not getting this permission:
![image](https://github.com/nextcloud/android/assets/1731941/5148555a-69c2-4eaa-8bf7-d404096ba54a)

**Comment by brilthor** [comment]
> @mrcg It already is. Well, it's requested at install time. The user can choose the other option on the pop-up though, which will obviously result in us not getting this permission

I just moved to a new phone, had to freshly set up next cloud client.  I did not receive this prompt.

I would also suggest the app should monitor to ensure it still has this permission before uploading corrupt data.

**Comment by joshtrichards** [comment]
@brilthor Are you by chance on Android OS <11?

**Comment by brilthor** [comment]
> @brilthor Are you by chance on Android OS <11?

Nope, Android 14 build UQ1A.231205.015

**Comment by vb0** [comment]
I think I mentioned it earlier, if the API that gives NextCloud zeroed out files has any return code that indicates it does that NextCloud should pay attention to it and notify quite prominently.   

If there is no indication about that NextCloud could still detect this from the content of the files (these files don't just have no GPS data, they have zeroed out GPS data). Or, easier, instead of trying to do something fancy just read the GPS via some standard library and if nothing sane is returned just warn on the transfer notification (with the option to disable the extra text somewhere in settings). This could help also if the location permission gets removed from the Camera app (because of inactivity) or some other shenanigans happen in the next flavor of Android and the user starts generating pictures without location, while actually wanting this data to be saved. For the users that want no GPS tags it shouldn't be too bad if they get this spelled out in the transfer notification (with the option to have the text removed and basically revert to what we (don't) do now), might actually be a welcome confirmation that they still have the desired configuration.

**Comment by TippyTurtle** [comment]
To "fix" (er work around) this on a Pixel 8 Pro running Android 14 do the following:
**Settings-->Apps-->Special App access (at the bottom)-->All Files Access-->Nextcloud-->Turn on "Allow access to manage all files"**
...in addition to the standard "App Permissions" being set correctly.

**Comment by cooper2k4** [comment]
Hi all,

sadly I'm now also facing this issue (missing GPS Exif tags in pictures uploaded through Nextcloud App). I'm using a Google Pixel 7a and got the Android 15 update recently. Maybe this has something to do with it. I cannot add the [above mentioned](https://github.com/nextcloud/android/issues/6248#issuecomment-2116029795) permission to the Nextcloud App. The Nextcloud App isn't in the list and can't be added to it.

**Comment by asm0dey** [comment]
Eventually I moved to immich for the photo storage. It's supreme in do many
aspects that I didn't even hesitate

**Comment by Cebrain** [comment]
> Hi all,
> 
> sadly I'm now also facing this issue (missing GPS Exif tags in pictures uploaded through Nextcloud App). I'm using a Google Pixel 7a and got the Android 15 update recently. Maybe this has something to do with it. I cannot add the [above mentioned](https://github.com/nextcloud/android/issues/6248#issuecomment-2116029795) permission to the Nextcloud App. The Nextcloud App isn't in the list and can't be added to it.

Same problem for me now, i used the workaround now for a long time. 
Nextcloud went now missing from the Special Apps permission.

Everyone, that had it working before please check your permission!!!

This is again a major problem, exact same problem as last time.
This setting also breaks now all backup jobs on your phone for file which are not Media Files!

For me the last day it worked was one 23rd September 2024

Check your files!

**Comment by Cebrain** [comment]
New Info, if you use the nextcloud application from the fdroid store, it will ask again for the permission.
Google did remove the permission. The fdroid version is working fine.

**Comment by DAveShillito** [comment]
Thanks to this thread I discovered I was losing the GPS data again, it seems to have occurred some time between the 14th & 18th December, so only a few files were affected and I could  sync them off the phone.

Is this a permission we can allow ourselves? or is switching to the fdroid version the only reliable workaround?

**Comment by dbmaxpayne** [comment]
I can confirm that it has also happened again to me on a Google Pixel 7 Pro.

**Comment by M3talGuy** [comment]
Yup, happened to me as well (Pixel 8 Pro). Synced Images up to Dec 14th still have GPS data, Images synced December 17th and later do not (no Images where taken on 15th and 16th). And yes, the NC App disappeared completely form the "Access all files" section , where i had set the option before as well.

**Comment by vb0** [comment]
> Eventually I moved to immich for the photo storage. It's supreme in do many aspects that I didn't even hesitate

This isn't much different for this type of issue, except for just being a different program that breaks at different times depending on who knows what Play Store policies, various OEM configurations, etc.

https://github.com/immich-app/immich/pull/2965
https://github.com/immich-app/immich/issues/13188
https://github.com/immich-app/immich/issues/11280
https://github.com/immich-app/immich/issues/11264

**Comment by AykutCevik** [comment]
This is so annoying... As others stated, it is happening again. The GPS data got zeroed-out. Not sure when it was the last time it worked and totally annoying to trace back the date and re-upload it again via a PC rather than using the app.

To be honest Nextcloud team, I don't get why this still is an issue and not handled with priority. This causes unrecoverable data loss and the problem is, that most of the users don't even know.

Nextcloud Android App Version: 3.30.7 from PlayStore.

**Comment by Defenso-QTH** [comment]
Since we cannot control Google's policies on the Play Store, I think the best workaround would be:
- Checking and displaying a warning on every upload if the app cannot read GPS info. (Warning that could be silenced in settings for users who do not care.)
- Enabling users to easily re-upload media that does not GPS info easily - or transfer only that information to update the files on the NextCloud remote.

**Comment by vb0** [comment]
No, this is not the problem, the OS is presenting a zeroed out file to NC. It doesn't matter what it does with it afterwards, the damage is done.

**Comment by Ra72xx** [comment]
Really? This would be deliberately destroying a file, which is IMHO unacceptable. This doesn't happen when copying the pictures as files with other options, e. g. FolderSync or any file explorer (at least I hope so). So I assumed that the Nextcloud app should simply access the pictures as files and not as media.

**Comment by vb0** [comment]
Everyone is in the same boat, hanging for dear life on the discussed permission. Sure, a file manager might have a lower chance to have it randomly yanked, doubly so if it's the default OS file manager, but there's nothing fundamentally different.

**Comment by andreaskoelsch** [comment]
I have the issue, too with a Pixel 8a. On the older phone, I could still set the full access permission in a previous nextcloud version. Even though this permission is no longer listed now, when uploading pictures with the old phone, they still come with GPS tags. On the new phone with the new nextcloud app, I could never set this permission, because it is no longer there. Uploads with the new phone are without GPS tags. Super annoying

**Comment by cooper2k4** [comment]
As far as I can see, the problem seems to be fixed now. A message appeared saying that I needed to check and grant full access permission, which I couldn't do, but it still works.

_Nextcloud Android App 3.31.0 on Pixel 7a_

**Comment by andreaskoelsch** [comment]
@cooper2k4 did you download the app from Play store or f-droid,

**Comment by cooper2k4** [comment]
> [@cooper2k4](https://github.com/cooper2k4) did you download the app from Play store or f-droid,

It is installed from Google Play Store.

**Comment by AykutCevik** [comment]
I can confirm that it works now on Android with version `3.31.0`.

**Comment by matezoltanfarkas** [comment]
I was having the same issue on my Pixel 8 Pro.
Downloading the F-Droid version fixed my issue.
weird indeed

**Comment by IT-Bob** [comment]
I've had the same problem for a few months now; I first noticed this issue in June 2026, both with auto-upload and manual uploads. I didn't have this problem before, and now I'm losing my location data.
I have a Pixel 7, on GrapheneOS (Android 16 then Android 17), Nextcloud from F-Droid.

**Comment by IT-Bob** [comment]
It seems the problem isn't actually related to Nextcloud after all. The only way I've found to preserve the GPS data is to connect my phone to my computer and copy the files. I lose this EXIF data if:
- I send the files via Nextcloud (even the F-Droid version);
- I copy them to an external hard drive connected to my phone;
- I copy them via File Explorer to Nextcloud (WebDAV) or via the DAVx5 WebDAV mount;
- I send the photos via KDEConnect.

**Comment by vb0** [comment]
@IT-Bob you are using an app for all the examples below, and the app doesn't have the "all files" permission mentioned multiple times in this long issue, for example like here: https://github.com/nextcloud/android/issues/6248#issuecomment-1207315271

"copy them to an external hard drive connected to my phone" is a bit weird if using the default file manager, as I'd guess it will get the permission to manage your files ... but nobody knows nowadays. Unfortunately this is a very "shifty" permission, that goes away sometimes randomly, or can't even be enabled, all depending on countless factors like the phone's manufacturer special tweaks, Android version, app version, where it's coming from (Play Store/F-Droid/etc.) and so on. It's kind of an ongoing battle just to be able to get your complete files from your own phone. The (few) ones that you can actually see in the regular shared storage, like the most basic camera pictures.

There are multiple ways to get the files via computer, like MTP, adb pull, probably Link to Windows. I'd say adb pull is most likely the safest for now.

**Comment by IT-Bob** [comment]
@vb0 Thanks, I missed those comments because they were hidden...
I had just granted the missing permissions _Photos and videos_ (I think it is this one, because KDEConnect do not have this option at all) and _Nearby devices_ and it works. 

https://community.e.foundation/t/preserving-exif-data-when-exporting-pictures/68434/2
